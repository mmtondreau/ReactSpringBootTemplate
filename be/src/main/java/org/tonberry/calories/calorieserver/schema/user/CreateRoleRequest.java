package org.tonberry.calories.calorieserver.schema.user;

import lombok.*;

import java.util.List;

@Setter
@Getter
@Builder(setterPrefix = "with")
@AllArgsConstructor
@NoArgsConstructor
public class CreateRoleRequest {
    private String name;
    private List<String> authorities;
}
