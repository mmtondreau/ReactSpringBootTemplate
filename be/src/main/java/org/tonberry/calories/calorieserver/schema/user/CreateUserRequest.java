package org.tonberry.calories.calorieserver.schema.user;

import lombok.*;

@Setter
@Getter
@Builder(setterPrefix = "with")
@AllArgsConstructor
@NoArgsConstructor
public class CreateUserRequest {
    private String username;
    private String password;
    private String apiKey;
    private String role;
}
