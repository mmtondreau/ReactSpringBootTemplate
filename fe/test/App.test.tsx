import { fireEvent, render, screen } from '@testing-library/react';
import user from '@testing-library/user-event';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import App from '../src/App';
import store from '../src/app/store';

jest.mock('../src/apollo-client.ts', () => {
  let count = 0;
  return {
    default: {
      query: jest.fn(() => Promise.resolve(count)),
      mutate: jest.fn(() => {
        count += 1;
        Promise.resolve(count);
      }),
    },
  };
});

// See https://reactjs.org/docs/testing-recipes.html
test('renders main page', async () => {
  render(
    <Provider store={store}>
      <App />
    </Provider>,
  );
  const increamentButton = await screen.findByText('Increment');
  expect(screen.getByText('count is: 0')).toBeInTheDocument();
  await act(() => user.click(increamentButton));
  expect(screen.getByText('count is: 1')).toBeInTheDocument();
});

test('snapshots match', async () => {
  const { getByText, asFragment } = render(
    <Provider store={store}>
      <App />
    </Provider>,
  );
  const firstRender = asFragment();
  fireEvent.click(getByText('Increment'));
  expect(firstRender).toMatchDiffSnapshot(asFragment());
});
