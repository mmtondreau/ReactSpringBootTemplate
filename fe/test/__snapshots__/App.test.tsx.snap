// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots match 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -19,11 +19,11 @@
              type=\\"button\\"
            >
              Increment
            </button>
            <span>
-             count is: -1
+             count is: 0
            </span>
            <button
              aria-label=\\"Decrement value\\"
              type=\\"button\\"
            >"
`;

exports[`snapshots match 2`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`snapshots match 3`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -19,11 +19,11 @@
              type=\\"button\\"
            >
              Increment
            </button>
            <span>
-             count is: -1
+             count is: -2
            </span>
            <button
              aria-label=\\"Decrement value\\"
              type=\\"button\\"
            >"
`;
